### PR TITLE
Add criteo training example

### PR DESCRIPTION
**Pull request by wjxiz1992**
- criteo training example shows the combination of NVTabular TensorFlow Horovod and Spark
- this example compared Petastorm vs NVTabular dataloader
- put sample data in small size inside the PR to help user validate quickly

Signed-off-by: wjxiz1992 <wjxiz1992@gmail.com>

### CODE REVIEW

**Comment by wjxiz1992** [comment]
@nvliyuan @GaryShen2008 Hi, I added a notebook for demo and also some markdown comments there, please help view and feedback! thanks!

File: examples/criteo_train/Criteo Training.ipynb [hunk 1]
```diff
@@ -0,0 +1,12650 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "66e38f1a",
+   "metadata": {},
+   "source": [
+    "### Build the model\n",
+    "\n",
+    "The model is compose by mainly 2 parts: embedding layers and fully-connected layers."
```
Comment (GaryShen2008) [inline examples/criteo_train/Criteo Training.ipynb:107]: compose => composed

Comment (wjxiz1992) [reply examples/criteo_train/Criteo Training.ipynb]: Corrected.

**Review by nvliyuan — COMMENTED** [review]
Names need to be consistent `tf-hvd-train-pure:0.1`

File: examples/criteo_train/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+We are using the preprocessed data from DLRM. All 40 columns(1 label + 39 features) are already numeric.
+
+In the following parts, we assume that the data are mounted as docker volumn at `/data/parquet`
+
+## Run Criteo example benchmark using Dockerfile
+
+1. Build the docker image
+```
+nvidia-docker build -f Dockerfile -t nvspark/tf-hvd-train:0.1 .
```
Comment (nvliyuan) [inline examples/criteo_train/README.md:15]: Names need to be consistent

Comment (wjxiz1992) [reply examples/criteo_train/README.md:15]: Updated.

File: examples/criteo_train/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+ENV PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION='python'
+
+
+RUN pip install pandas sklearn ortools nvtx-plugins pydot && \
+    pip cache purge
+
+# tf-nightly for performance test
+# more details: https://github.com/tensorflow/tensorflow/issues/44194
+RUN pip uninstall tensorflow -y; pip install tf-nightly==2.7.0.dev20210722
```
Comment (nvliyuan) [inline examples/criteo_train/Dockerfile:150]: Do we need to change tf-nightly to tensorflow?

Comment (wjxiz1992) [reply examples/criteo_train/Dockerfile:150]: Updated but re-install keras-nightly with specific version.

**Comment by nvliyuan** [comment]
Should we add a description in README.md about this example? So that customers will know it on the front page.

File: examples/criteo_train/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,213 @@
+ARG IMAGE=nvcr.io/nvidia/tensorflow:21.03-tf2-py3
```
Comment (GaryShen2008) [inline examples/criteo_train/Dockerfile:1]: CopyRight?

File: examples/criteo_train/README.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
```
Comment (GaryShen2008) [inline examples/criteo_train/README.md:17]: Do we need nvidia-docker to build dockerfile?

File: examples/criteo_train/README.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+--conf spark.executor.resource.gpu.amount=1 \
+--conf spark.executor.resource.gpu.discoveryScript=./getGpusResources.sh \
+--files $SPARK_HOME/examples/src/main/scripts/getGpusResources.sh
+
+```
+
+
+## Note: 
+
+if you only want to try in a node with only 1 GPU, please modify the GPU number per worker in `$SPARK_HOME/conf/spark-env.sh` before you launch spark workers becasue the docker image is targeted for DGX-2 with 16 GPUs
```
Comment (GaryShen2008) [inline examples/criteo_train/README.md:71]: Nit: if => If

File: examples/criteo_train/criteo_keras.py [hunk 7]
```diff
@@ -0,0 +1,293 @@
+import argparse
```
Comment (GaryShen2008) [inline examples/criteo_train/criteo_keras.py:1]: Copyright?

**Review by GaryShen2008 — CHANGES_REQUESTED** [review]

File: examples/criteo_train/README.md (hunk 5 above)
Comment (wjxiz1992) [reply examples/criteo_train/README.md:17]: Yes, there're also Tensorflow and horovod installation in the Dockerfile, and they will require GPU to do check according to my test.

File: examples/criteo_train/Dockerfile (hunk 4 above)
Comment (wjxiz1992) [reply examples/criteo_train/Dockerfile:17]: Added.

File: examples/criteo_train/README.md (hunk 6 above)
Comment (wjxiz1992) [reply examples/criteo_train/README.md]: Updated.

File: examples/criteo_train/criteo_keras.py (hunk 7 above)
Comment (wjxiz1992) [reply examples/criteo_train/criteo_keras.py:17]: Added

**Comment by wjxiz1992** [comment]
> Should we add a description in README.md about this example? So that customers will know it on the front page.

Added. @nvliyuan

**Comment by wjxiz1992** [comment]
Hi, can I get another round review for the comments? thx!

**Pull request merged**